### PR TITLE
noetic-devel

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

File: src/grammar_parser/cfgparser.py [hunk 1]
```diff
@@ -85,7 +91,7 @@ class bcolors:
     UNDERLINE = '\033[4m'
 
 
-class Option:
+class Option(object):
```
Comment (ar13pit) [inline src/grammar_parser/cfgparser.py:94]: Don't need this.

File: src/grammar_parser/cfgparser.py [hunk 2]
```diff
@@ -150,9 +160,7 @@ def to_graphviz(self, graph):
 # ----------------------------------------------------------------------------------------------------
 
 
-
-
-class Conjunct:
+class Conjunct(object):
```
Comment (ar13pit) [inline src/grammar_parser/cfgparser.py:163]: Don't need this

File: src/grammar_parser/cfgparser.py [hunk 3]
```diff
@@ -189,14 +198,14 @@ def to_graphviz(self, graph):
 # ----------------------------------------------------------------------------------------------------
 
 
-
-class Rule:
+class Rule(object):
```
Comment (ar13pit) [inline src/grammar_parser/cfgparser.py:201]: Don't need this

File: src/grammar_parser/cfgparser.py [hunk 4]
```diff
@@ -238,8 +248,7 @@ def to_graphviz(self, graph):
 # ----------------------------------------------------------------------------------------------------
 
 
-
-class Tree:
+class Tree(object):
```
Comment (ar13pit) [inline src/grammar_parser/cfgparser.py:251]: Don't need this

File: src/grammar_parser/cfgparser.py [hunk 5]
```diff
@@ -297,20 +305,20 @@ def parse_next_atom(s):
...
 
-    return (s, "", "")
+    return s, "", ""
 
 
 # ----------------------------------------------------------------------------------------------------
 
 
-class CFGParser:
+class CFGParser(object):
```
Comment (ar13pit) [inline src/grammar_parser/cfgparser.py:321]: Don't need this

File: CMakeLists.txt [hunk 6]
```diff
@@ -27,10 +27,10 @@ catkin_python_setup()
 ##   your messages/services/actions (e.g. std_msgs, actionlib_msgs, ...).
 ## * In the file package.xml:
 ##   * add a build_depend tag for "message_generation"
-##   * add a build_depend and a run_depend tag for each package in MSG_DEP_SET
+##   * add a build_depend and a exec_depend tag for each package in MSG_DEP_SET
```
Comment (ar13pit) [inline CMakeLists.txt:30]: Maybe delete these comments all together

**Review by ar13pit — DISMISSED** [review]

File: src/grammar_parser/cfgparser.py (hunk 1 above)
Comment (MatthijsBurgh) [reply src/grammar_parser/cfgparser.py:94]: But, why not? Do you want me to remove it?

Comment (ar13pit) [reply src/grammar_parser/cfgparser.py:94]: The inheritance from object is default in python3. This syntax is only used for backwards compatibility with python2.

Comment (MatthijsBurgh) [reply src/grammar_parser/cfgparser.py:94]: So this is an improvement. So that it behaves the same in python2 and python3

Comment (ar13pit) [reply src/grammar_parser/cfgparser.py:94]: But we are moving to python3, lets not support python2 anymore. Its a hassle maintaining a deprecated version.

Comment (MatthijsBurgh) [reply src/grammar_parser/cfgparser.py:94]: I prefer compatibility for both for a moment. But for this one we can remove it.

File: src/grammar_parser/cfgparser.py [hunk 7]
```diff
@@ -630,19 +644,22 @@ def has_completion_function(self, func_name):
...
+        for name, rule in self.rules.items():
             graph.edge(self.graphviz_id(), rule.graphviz_id())
             rule.to_graphviz(graph)
 
     def visualize_options(self, graph, target_rule, previous_words=None, depth=2):
         previous_words = [] if not previous_words else previous_words
 
         import itertools
-        colors = itertools.cycle(["blue", "green", "red", "cyan", "magenta", "black", "purple", "orange"])
+        colors = itertools.cycle(
```
Comment (jlunenburg) [inline src/grammar_parser/cfgparser.py:660]: Why is itertools imported here?

File: src/grammar_parser/cfgparser.py [hunk 8]
```diff
@@ -653,15 +670,18 @@ def visualize_options(self, graph, target_rule, previous_words=None, depth=2):
         next_words = set(self.next_word(target_rule, previous_words))
 
         if next_words and depth:
-            # print next_words
+            # print(next_words)
```
Comment (jlunenburg) [inline src/grammar_parser/cfgparser.py:673]: Just remove?

File: src/grammar_parser/cfgparser.py [hunk 9]
```diff
@@ -126,10 +133,11 @@ def from_cfg_def(option_definition, left_semantics):
             yield opt
 
     def pretty_print(self, level=0):
-        # print self, level
+        # print(self, level)
```
Comment (jlunenburg) [inline src/grammar_parser/cfgparser.py:136]: Just remove/

**Review by jlunenburg — CHANGES_REQUESTED** [review]
Nice that you've also cleaned it up a little. I added some minor suggestions (maybe remove other deprecated code as well).

**Review by jlunenburg — APPROVED** [review]

**Pull request merged**